Vendor collections for isolated jobs to work in ansible 2.10

fosterseth opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
#6930 

install ansible modules necessary to run isolated jobs / container groups

<!---
If you are fixing an existing issue, please include "related #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Bugfix Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
 - API
 - Installer

##### AWX VERSION
<!--- Paste verbatim output from `make VERSION` between quotes below -->
```
awx: 11.2.0

```

AlanCoding commented on requirements/collections_requirements.yml:
```diff
@@ -16,3 +16,7 @@ collections:
+  - name: community.kubernetes
+    version: 0.11.0
+  - name: ansible.posix
+    version: 0.1.0
```
let's comment that these are required for the isolated management playbooks.

I think it makes sense to combine them right now - but they will be separated back out as execution environments start to be used for inventory imports.

fosterseth replied:
left comments

AlanCoding reviewed (approved):
if it works, I would go for it

shanemcd commented on awx/main/isolated/manager.py:
```diff
@@ -74,6 +74,7 @@ def build_runner_params(self, hosts, verbosity=1):
         env['ANSIBLE_RETRY_FILES_ENABLED'] = 'False'
         env['ANSIBLE_HOST_KEY_CHECKING'] = str(settings.AWX_ISOLATED_HOST_KEY_CHECKING)
         env['ANSIBLE_LIBRARY'] = os.path.join(os.path.dirname(awx.__file__), 'plugins', 'isolated')
+        env['ANSIBLE_COLLECTIONS_PATHS'] = settings.INVENTORY_COLLECTIONS_ROOT
```
We should rename this variable now that it's being used for more than just inventory plugins.

AlanCoding replied:
It's in the name of the folder too, which seems like a bigger problem than the setting name.

My thinking is these will be replaced before it really matters.

shanemcd replied:
Renaming a few things isn't hard. We dont know if / when this vendoring solution is going away. Shouldn't we spend a few extra seconds to make things make sense?

ryanpetrello replied:
Why not just make the variable `settings.AWX_ANSIBLE_COLLECTIONS_PATHS` so it's super clear what it represents?

fosterseth replied:
going with Ryan's suggestion, thanks!

shanemcd reviewed (commented):
LGTM, but please rebase / squash some commits before merging this.

AlanCoding reviewed (approved):
once the tests show that it's working, then go ahead and get it moving

thanks for doing it!

fosterseth committed:
Vendor collections for isolated jobs to work in ansible 2.10

kubectl and synchronize are now part of community.kubernetes
and ansible.posix collections, respectively. This change installs
these collections to a local directory to be used in inventory and
isolated management playbooks.

awx issue #6930

fosterseth commented:
regate

fosterseth commented:
recheck

fosterseth commented:
recheck

fosterseth commented:
recheck

fosterseth committed:
flake 8 version test
